### PR TITLE
Mo homework

### PR DESCRIPTION
**Pull request by Artificial-Friend**
Master.java - start of the program.

### CODE REVIEW

File: src/main/java/core/basesyntax/Figure.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public abstract class Figure {
+
+    public double area;
+    public String color;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Figure.java:6]: let's change the access modifier from `public` to `private`

File: src/main/java/core/basesyntax/Figure.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    public Figure(double area, String color) {
+        this.area = area;
+        this.color = color;
+    }
+
+    public abstract void draw();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Figure.java:13]: this method better to move to the interface:
* interface - behavior
* abstract class - state

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureGenerator {
+
+    private static final String[] colors = new String[]{"black", "white", "whatever"};
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:7]: we can use `enum` for colors - you can read about it later
But now - `Srting array` is enough :)

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    private static final String[] figuresArray = new String[]{
+            "circle", "right triangle", "rectangle", "square", "isosceles trapezoid"};
+
+    public static Figure create() {
+        switch (figuresArray[new Random().nextInt(5)]) {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:12]:
```suggestion
        switch (figuresArray[new Random().nextInt(figuresArray.length)]) {
```

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            case "rectangle":
+                return new Rectangle(getRandomColor(), getRandomNumber(), getRandomNumber());
+            case "square":
+                return new Square(getRandomColor(), getRandomNumber());
+            case "isosceles trapezoid":
+                return new IsoscelesTrapezoid(getRandomColor(), getRandomNumber(),
+                                                getRandomNumber(), getRandomNumber());
+            default:
+                System.exit(-1);
+                return new Circle(getRandomColor(), getRandomNumber());
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:26]: now we can return `null` here

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        }
+    }
+
+    public static String getRandomColor() {
+        return colors[new Random().nextInt(3)];
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:31]:
```suggestion
        return colors[new Random().nextInt(colors.length)];
```

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+            case "isosceles trapezoid":
+                return new IsoscelesTrapezoid(getRandomColor(), getRandomNumber(),
+                                                getRandomNumber(), getRandomNumber());
+            default:
+                System.exit(-1);
+                return new Circle(getRandomColor(), getRandomNumber());
+        }
+    }
+
+    public static String getRandomColor() {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:30]: let's make this method `private`

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:4]: let's remove empty line 
check the same in all classes))

File: src/main/java/core/basesyntax/Master.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+public class Master {
+
+    public static void main(String[] args) {
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Master.java:6]: let's remove empty line

File: src/main/java/core/basesyntax/Master.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+        Figure[] randomArrayOfRandomFigures = new Figure[FigureGenerator.getRandomNumber()];
+
+        for (int i = 0; i < randomArrayOfRandomFigures.length; i++) {
+            randomArrayOfRandomFigures[i] = FigureGenerator.create();
+            randomArrayOfRandomFigures[i].draw();
+        }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Master.java:13]: remove empty line

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureGenerator.java (hunk 3 above)
Comment (Artificial-Friend) [reply src/main/java/core/basesyntax/FigureGenerator.java]: Your wish is my command

File: src/main/java/core/basesyntax/FigureGenerator.java (hunk 4 above)
Comment (Artificial-Friend) [reply src/main/java/core/basesyntax/FigureGenerator.java]: Obviously smart - done!

File: src/main/java/core/basesyntax/FigureGenerator.java (hunk 5 above)
Comment (Artificial-Friend) [reply src/main/java/core/basesyntax/FigureGenerator.java]: I should have seen that. Done!

File: src/main/java/core/basesyntax/Figure.java (hunk 2 above)
Comment (Artificial-Friend) [reply src/main/java/core/basesyntax/Figure.java]: It seems my brain was melted yesterday - done!

**Comment by Artificial-Friend** [comment]
Edited all empty lines.

File: src/main/java/core/basesyntax/FigureGenerator.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    private static Colors getRandomColor() {
+        return Colors.values()[new Random().nextInt(Colors.values().length)];
+    }
+
+    public static int getRandomNumber() {
+        return new Random().nextInt(200);
+    }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FigureGenerator.java:35]: we can remove one empty line

**Review by Sofasmile — APPROVED** [review]
good job

**Pull request closed without merging**